### PR TITLE
generic delegate macros

### PR DESCRIPTION
**Pull request by wash2**
This PR adds generic delegate macros to sctk. See https://github.com/wash2/client-toolkit/blob/master/examples/generic_list_seats.rs for an example.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -23,18 +23,23 @@ lazy_static = "1.0"
...
+wayland-protocols-wlr = { git="https://github.com/smithay/wayland-rs", features = ["client"] }
+wayland-cursor = { git="https://github.com/smithay/wayland-rs" }
 # Explicit dependency until release
 xkbcommon = { git = "https://github.com/rust-x-bindings/xkbcommon-rs", optional = true, features = ["wayland"] }
 calloop = { version = "0.10.0", optional = true }
 
 [features]
-default = ["calloop", "xkbcommon"]
+default = ["calloop", "xkbcommon", "system"]
+system = ["wayland-backend/client_system"]
```
Comment (vberger) [inline Cargo.toml:42]: Why add this feature, and why make it default?

Also this looks unrelated to this PR and should be discussed in a separate issue/PR imo.

Comment (wash2) [reply Cargo.toml:42]: I've been using as_ptr() for a while and recently needed to add this feature to make the method available again. I figured all features were enabled by default, so I enabled it by default. I can move it to a different PR though.

Comment (vberger) [reply Cargo.toml]: Mostly the encouraged method to use `as_ptr()` for a project is by itself depending directly on `wayland-backend` to activate the feature, as this method is part of `wayland-backend` API. So I don't see any reason to add that feature to SCTK, given it has 0 impact on the crate.

Comment (wash2) [reply Cargo.toml]: ok that change has been removed

**Comment by wash2** [comment]
I've updated to resolve merge conflicts, but beta.9 still seems to break the examples for me, so I've left the git dependencies.

**Comment by wash2** [comment]
I've updated to use the latest beta and squashed my changes.

**Review by i509VCB — APPROVED** [review]
